unit-tests-java8/src/test/java/org/eclipse/collections/test/stack/StackIterableTestCase.java

```diff
@@ -60,6 +89,143 @@ default void InternalIterable_forEach()
         assertEquals(this.newWith(1, 2, 2, 3, 3, 3), result);
     }
 
+    @Override
+    default void RichIterable_flatCollectBoolean()
+    {
+        RichIterable<Twin<Boolean>> booleans = this.newWith(Tuples.twin(true, true), Tuples.twin(false, false));
+        BooleanArrayList target = new BooleanArrayList();
+        booleans.flatCollectBoolean(new Function<Twin<Boolean>, BooleanIterable>()
+        {
+            @Override
+            public BooleanIterable valueOf(Twin<Boolean> each)
+            {
+                return BooleanArrayList.newListWith(each.getOne(), each.getTwo());
+            }
+        }, target);
+        assertSize(4, target);
```

Comment:
Use an equality test with the target which covers size and containsAll.

Reply:
Same pattern here as well with "target" and "result".